### PR TITLE
ci: add more revive linters

### PR DESCRIPTION
**Pull request by ccoVeille**
- **ci: enable revive.deep-exit and fix reported errors**
- **ci: enable revive.unused-receiver and fix reported errors**

Related to #787

### CODE REVIEW

**Review by vipul-rawat — DISMISSED** [review]

**Comment by ccoVeille** [comment]
Strange to see the tests are now broken.

I'll have a look

**Comment by vipul-rawat** [comment]
> Strange to see the tests are now broken.
> 
> I'll have a look

Reruning, SQL container not reachable error! Need to take a look, why this is happening.

File: pkg/gofr/logging/logger.go [hunk 1]
```diff
@@ -32,8 +32,6 @@ type Logger interface {
 	Warnf(format string, args ...interface{})
 	Error(args ...interface{})
 	Errorf(format string, args ...interface{})
-	Fatal(args ...interface{})
-	Fatalf(format string, args ...interface{})
```
Comment (srijan-27) [inline pkg/gofr/logging/logger.go:36]: @ccoVeille - why was this removed?

Comment (ccoVeille) [reply pkg/gofr/logging/logger.go:36]: Apparently, mothing was using it, and one of them implementation was doing a deep exit:

Calling a os.Exit(1) in the code, slaughtering all the stacks and the possible defer, context cancelation.

Did I miss something and it was called?
